jekyll/_cci2/build-publish-snap-packages.md

```diff
@@ -0,0 +1,177 @@
+          command: |
+            mkdir .snapcraft
+            echo $SNAPCRAFT_LOGIN_FILE | base64 --decode --ignore-garbage > .snapcraft/snapcraft.cfg
+            snapcraft push *.snap --release stable
+...
+```
+
+2. Once the base64 encoded version of the file is stored on CircleCI as a private environment variable, we can then use it within a build to automatically publish to the store.
+
+In this example, Snapcraft automatically looks for login credentials in `.snapcraft/snapcraft.cfg` so we decode the environment variable we made previously right into that location. `snapcraft push` is then used to upload the .snap file into the Snap Store.
+
+### Uploading vs Releasing
+
+`snapcraft push *.snap` by default will upload the snap to the Snap Store, run any store checks on the server side, and then stop. The snap won't be "released" meaning users won't automatically see the update. The snap can be published locally with the `snap release <release-id>` command or by logging into the Snap Store and clicking the release button.
+
+In typical CircleCI fashion, we can go fully automated (as in the above example) but using the `--release <channel>` flag. This uploads the snap, does Store side verification, and then will automatically release the snap in the specified channels. 
+
+
+## Workflows
+
+We can utilize multiple jobs to better organize our snap build. A job to build/compile the actual project, a job to build the snap itself, and a job that published the snap (and other packages) only on `master` would all be useful.
+
+[Workflows][workflows] can help with building snaps in two ways:
+
+1. **Snap Store Channels** - As we mentioned in the previous section, when we upload to the Store we could optionally release at the same time. This allows us to designate specific jobs on CircleCI to deploy to specific Snap Channels. For example, the `master` branch could be used to deploy to the `edge` channel` while tagged releases could be used to deploy to the `stable` channel.
+1. **Parallelize Packing** - If your software is being packaged as a snap as well as something else, say a flatpak, .deb, .apk, etc, each package type could be placed in its own job and all run parallel. This allows your build to complete must fast than if the .deb package could start to build until the snap completed, and so on.
+
+Utilize CircleCI `workspaces` to move a generated snap file between jobs when neccessarily. Here's an example showing a snippet from the "from" job and a snippet of the "to" job:
+
+```YAML
+... # from a job that already has the snap
+      - persist_to_workspace:
+          root: .
+          paths:
+            - "*.snap"
+... # to the next job that needs the snap
+      - attach_workspace:
+          at: .
+...
+```
+
+Below is a complete example of how a snap package could be built on CircleCI. This same process is used the build the Snap pakcage for the [CircleCI Local CLI][local-cli-repo].
+
+
+## Full Example Config
```

Comment:
Please comment to each main stanza of this config with what it is doing.

Reply:
What would the comments say? The comments is the whole doc.